### PR TITLE
FAI-10447: CircleCI fix incremental sync

### PR DESCRIPTION
**Pull request by chalenge**
## Description

- CircleCI fix incremental sync, make project slugs consistent and incorrect lastUpdated
- Fix block_list param inconsistency

## Type of change
- [X] Bug fix
- [ ] New feature
- [ ] Breaking change

### CODE REVIEW

File: sources/circleci-source/src/circleci/circleci.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ export interface CircleCIConfig {
   readonly token: string;
   readonly url?: string;
   project_slugs: ReadonlyArray<string>;
-  readonly project_blocklist?: string[];
+  readonly project_block_list?: string[];
```
Comment (chalenge) [inline sources/circleci-source/src/circleci/circleci.ts:24]: Matches [here](https://github.com/faros-ai/airbyte-connectors/blob/main/sources/circleci-source/resources/spec.json#L48)

**Review by matiaslcoulougian — APPROVED** [review]

File: sources/circleci-source/resources/spec.json [hunk 2]
```diff
@@ -43,15 +43,15 @@
...
+        "description": "Project slugs look like \"vcs-slug/org-name/repo-name\". Enter \"*\" for all projects."
       },
       "project_block_list": {
         "type": "array",
         "items": {
           "type": "string"
         },
         "title": "Project Slug Blocklist",
-        "description": "Project slugs should look like \"vcs-slug/org-name/repo-name\""
+        "description": "Project slugs look like \"vcs-slug/org-name/repo-name\" or with wildcards like \"vcs-slug/org-name/*\""
```
Comment (ypc-faros) [inline sources/circleci-source/resources/spec.json:54]: Same here, add example(s)

File: sources/circleci-source/src/circleci/circleci.ts [hunk 3]
```diff
@@ -151,17 +151,23 @@ export class CircleCI {
         // //circleci.com/organization_id/project_id
         // https://github.com/organization/repository
         const projectMatch = projectVCSUrl.match(
-          /^[^/]*\/\/([^./]+).*\/([^./]+)\/([^./]+)$/
+          /^[^/]*\/\/([^./]+).*\/([^./]+)\/([\w.-]+)$/
         );
         if (!projectMatch || projectMatch.length < 1) {
```
Comment (ypc-faros) [inline sources/circleci-source/src/circleci/circleci.ts:156]: Below we do `const projectId = projectMatch[3];` so should this assert for length >= 4?

Comment (chalenge) [reply sources/circleci-source/src/circleci/circleci.ts:156]: Oh yeah, I was curios about that. Forgot to ask @willmarks why we only check for < 1?

File: sources/circleci-source/src/circleci/circleci.ts [hunk 4]
```diff
@@ -280,7 +299,7 @@ export class CircleCI {
     startDate.setDate(startDate.getDate() - this.cutoffDays);
 
     const lastUpdatedAt = since ? new Date(since) : startDate;
-    const gracePeriod = lastUpdatedAt;
+    const gracePeriod = new Date(lastUpdatedAt);
```
Comment (ypc-faros) [inline sources/circleci-source/src/circleci/circleci.ts:302]: It was already a `Date` ...

File: sources/circleci-source/resources/spec.json (hunk 2 above)
Comment (chalenge) [reply sources/circleci-source/resources/spec.json:54]: @ypc-faros. I was worried about the size of the catalog or this doesn't apply here?

Comment (ypc-faros) [reply sources/circleci-source/resources/spec.json:54]: No, it's only for the Faros destination spec 👍

File: sources/circleci-source/src/circleci/circleci.ts (hunk 4 above)
Comment (chalenge) [reply sources/circleci-source/src/circleci/circleci.ts:302]: This was a gotcha. Not sure why we need the gracePeriod but if don't make it a new Date reference gracePeriod points to same reference as lastUpdatedAt. It doesn't get the value only, thus when we update [gracePeriod value](https://github.com/faros-ai/airbyte-connectors/pull/1414/files#diff-dcadf7d560989bc22d562a359cb75ca4284aee3713bb6f5166275f5a4c7c8beaL284) lastUpdated also points to this new updated value resulting in the [syncs always pulling data ](https://github.com/faros-ai/airbyte-connectors/pull/1414/files#diff-dcadf7d560989bc22d562a359cb75ca4284aee3713bb6f5166275f5a4c7c8beaL303)from wayback 🤦🏽 
image
what we want is two different values
image

Can put a comment

Comment (ypc-faros) [reply sources/circleci-source/src/circleci/circleci.ts:302]: Omg 🤯 
`Not sure why we need the gracePeriod` <= yeah good point, maybe we could get rid of it altogether? It doesn't seem like `lastUpdatedAt` is updated anywhere ... 

We should clean up this logic (maybe not now...) It's hard to read for me

**Review by ypc-faros — APPROVED** [review]

File: sources/circleci-source/src/circleci/circleci.ts (hunk 4 above)
Comment (chalenge) [reply sources/circleci-source/src/circleci/circleci.ts:302]: Yeah, we should update it, but needs more understanding of how CircleCi works
We use `lastUpdatedAt` to filter in `fetchWorkflows` and use gracePeriod for `pipelines`

File: sources/circleci-source/src/circleci/circleci.ts (hunk 3 above)
Comment (willmarks) [reply sources/circleci-source/src/circleci/circleci.ts]: The < 1 wasn't really necessary iirc as I think it is redundant with checking that projectMatch exists but I added per a review comment. Something like `!projectMatch?.length` probably would have been better.

**Pull request merged**